src/components/Modal.tsx

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+
+interface ModalProps {
+    heading: string;
+    body: JSX.Element;
+    setShow: (set: boolean) => void;
+    show: boolean;
+}
+
+function Modal(props: ModalProps) {
+
+    let modalClass: string = "modal";
+    if (props.show) modalClass = "modal is-active";
+
+    return (
+    <div>
+        <div className = {modalClass}>
+            <div className = "modal-background"></div>
+            <div className = "modal-card">
+                <header className = "modal-card-head has-text-centered">
+                    <h1 className = "modal-card-title">{props.heading}</h1>
+                    <button className = "delete" onClick = {() => {props.setShow(false);}}></button>
+                </header>
+                <div className = "modal-card-body">
+                    {props.body}
```

Comment:
Here you could use `props.children` which lets you do something like:
```js
<Modal heading="Hello World">
  content here
</Modal>
```

You'll have to switch `body` in the interface to `JSX.Element[]`

Reply:
Yeah, that would keep it much cleaner and easier to use! Making the necessary changes now...